app/forms/applicants/previous_reference_form.rb

```diff
@@ -0,0 +1,34 @@
+module Applicants
+  class PreviousReferenceForm < BaseForm
+    form_for Applicant
+
+    CCMS_REFERENCE_REGEXP = /\A[13][0-9]{11}\z/
+
+    attr_accessor :applied_previously, :previous_reference
+
+    before_validation :normalise_previous_reference
+
+    validates :applied_previously, inclusion: %w[true false], unless: :draft?
+    validates :previous_reference, presence: true, if: :applied_previously?, unless: :draft?
+    validate :validate_previous_reference, unless: :draft?
+
+  private
+
+    def applied_previously?
+      applied_previously.eql?("true")
+    end
+
+    def normalise_previous_reference
+      attributes[:previous_reference] = nil unless applied_previously?
+      return if previous_reference.blank?
+
+      previous_reference.delete!(" ")
```

Comment:
@skdejong it would be good to add a test for this to your form spec, to test that if the add a previous application reference with a space, say "321 123456789" it is saved as "321123456789"

Reply:
Thank you for the suggestion. I have added this test now to the form spec.